File: Level2/221101/JavaScript/Carpet.js
export function squareRoot(brown, yellow) {
  return ((2 + brown / 2)
  + Math.sqrt((2 + brown / 2)
  ** 2 - 4 * (brown + yellow))) / 2;
}

export function solution(brown, yellow) {
  const x = squareRoot(brown, yellow);
  const y = brown / 2 + 2 - x;
  return [x, y];
}

File: Programmers/Level2/221031/Carpet.java
public class Carpet {
  public int[] solution(int brown, int yellow) {
    int sum = sum(brown, yellow);

    return factorization(sum, brown);
  }

  public int sum(int brown, int yellow) {
    return brown + yellow;
  }

  public int[] factorization(int sum, int brown) {
    int[] factorization = new int[2];
    for (int i = 2; i < sum; i += 1) {
      if (sum % i == 0) {
        int maximumNumber = sum / i;
        if ((i * 2 + maximumNumber * 2 - 4) == brown) {
          factorization[0] = i;
          factorization[1] = maximumNumber;
        }
      }
    }
    return factorization;
  }
}
